src/pk/ecc/ecc_set_dp.c

```diff
@@ -37,9 +37,32 @@ int ecc_set_dp(const ltc_ecc_curve *curve, ecc_key *key)
    /* cofactor & size */
    key->dp.cofactor = curve->cofactor;
    key->dp.size = mp_unsigned_bin_size(key->dp.prime);
-   /* OID */
-   key->dp.oidlen = curve->oidlen;
-   for (i = 0; i < key->dp.oidlen; i++) key->dp.oid[i] = curve->oid[i];
+   /* OID string >> unsigned long oid[16] + oidlen */
+   if (curve->OID != NULL) {
+      for (i = 0, j = 0; i < strlen(curve->OID); i++) {
+         if (curve->OID[i] == '.') {
```

Comment:
how about adding `str_get_oid(const char*, oid_st*)` similar to `pk_get_oid()`?

Reply:
...which converts the input, not a LUT as the latter... :) and `str_to_oid(const char*, unsigned char*, unsigned long*)` would also be perfectly fine!

Reply:
done, see pk_oid_str_to_num + pk_oid_num_to_str